test/unit/datasets/test_datasets.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import pandas as pd
+import numpy as np
+
+from fairlearn.datasets import fetch_adult, fetch_boston
+
+# =============================================
+
+
+class TestFairlearnDataset:
+    def test_dataset_as_bunch_in_pandas_format(self):
+        fetch_function_list = [fetch_adult, fetch_boston]
```

Comment:
Could you make these into [data-driven tests](https://docs.pytest.org/en/stable/parametrize.html), so that an early failure won't prevent testing of the other `fetch` functions.

Reply:
Thanks for the suggestion. Added the parameterization via pytest.

Reply:
I think @riedgar-ms meant that fetch_adult and fetch_boston should also be parametrized.

Reply:
Yes, that's actually what I was thinking of, although keeping the "as_frame" parameterisation also looks like it makes sense. An example of parameterising over a list of functions:
https://github.com/fairlearn/fairlearn/blob/8f087fbb0b27740d10b31d95706bb175a4b4581c/test/unit/metrics/test_metrics_engine.py#L31